### PR TITLE
Misc doc changes

### PR DESCRIPTION
**Pull request by kianmeng**
Besides other documentation changes, this commit ensures the generated
HTML doc for HexDocs.pm will become the main source doc for this Elixir
library which leverage on latest ExDoc features.

### CODE REVIEW

File: lib/logger_json/formatters/datadog_logger.ex [hunk 1]
```diff
@@ -1,6 +1,8 @@
 defmodule LoggerJSON.Formatters.DatadogLogger do
   @moduledoc """
-  DataDog formatter. Adhears to the DataDog
+  DataDog formatter.
+
+  Adhere to the Doted
```
Comment (AndrewDryga) [inline lib/logger_json/formatters/datadog_logger.ex:5]:
```suggestion
  Adhere to the
```

File: lib/logger_json/formatters/google_cloud_logger.ex [hunk 2]
```diff
@@ -2,6 +2,7 @@ defmodule LoggerJSON.Formatters.GoogleCloudLogger do
   @moduledoc """
   Google Cloud Logger formatter.
   """
+
```
Comment (AndrewDryga) [inline lib/logger_json/formatters/google_cloud_logger.ex:5]: We don't add new line between moduledoc and `alias/import/include` to make the header smaller.

File: lib/logger_json/formatters/google_error_reporter.ex [hunk 3]
```diff
@@ -1,4 +1,8 @@
 defmodule LoggerJSON.Formatters.GoogleErrorReporter do
+  @moduledoc """
+  Google Error Reporter formatter.
+  """
+
```
Comment (AndrewDryga) [inline lib/logger_json/formatters/google_error_reporter.ex:5]:
```suggestion
```

File: lib/logger_json/plug.ex [hunk 4]
```diff
@@ -3,6 +3,7 @@ if Code.ensure_loaded?(Plug) do
     @moduledoc """
     A Plug to log request information in JSON format.
     """
+
```
Comment (AndrewDryga) [inline lib/logger_json/plug.ex:6]:
```suggestion
```

**Comment by AndrewDryga** [comment]
Hey @kianmeng, I've left a couple of small nitpicks and LGTM. Thank you :).

File: mix.exs [hunk 5]
```diff
@@ -1,72 +1,72 @@
...
-  #
-  # Type "mix help deps" for more examples and options
   defp deps do
     [
       {:jason, "~> 1.0"},
       {:ecto, "~> 2.1 or ~> 3.0", optional: true},
       {:plug, "~> 1.0", optional: true},
       {:telemetry, "~> 0.4.0", optional: true},
-      {:ex_doc, ">= 0.15.0", only: [:dev, :test]},
+      {:ex_doc, ">= 0.0.0", only: [:dev, :test], runtime: false},
```
Comment (AndrewDryga) [inline mix.exs:39]: I'm not sure why would we drop the version requirement here? There is no way how we would be downgraded to 0.14.X or earlier anyways. So let's drop the version match change and just add the `runtime: false`.

**Comment by kianmeng** [comment]
> Hey @kianmeng, I've left a couple of small nitpicks and LGTM. Thank you :)

@AndrewDryga Noted and requested changes amended. :smile:

**Comment by AndrewDryga** [comment]
Thank you!

**Pull request merged**